app/routes/analyzer.tsx

```diff
@@ -735,6 +751,88 @@ function AbilityPointsDetails({
   );
 }
 
+/**
+ * From an array of Main abilities, create a map of <Ability, number>, then return it as an Array after sorting by value, descending.
+ *
+ * The map describes the number of Ability chunks required to replace the main ability on a piece of gear.
+ *
+ * Extra processing is required for Main abilities that are primary slot-only abilities,
+ *  as they are comprised of 3 stackable ability chunks at a lower ability chunk count than usual.
+ */
+function getAbilityChunksMapAsArray(mainAbilities: AbilityWithUnknown[]) {
```

Comment:
Let's move this function to the analyzer module (new file) + maybe add test or two for it.

Reply:
Great idea, I forgot `.ts` is supposed to contain more detailed/in-depth logic & `.tsx` is supposed to contain DOM-specific logic for the most part.

I did this & am in the process of adding tests

Reply:
@Sendouc I added some meaningful tests just now - can you please take a look when you have time? Thanks =)

Reply:
Updated tests again btw (they are really good now)